ikasaneip/module/src/main/java/org/ikasan/module/service/StartupModuleProcessor.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.util.CollectionUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Uses StartupModuleConfiguration to perform module configuration actions at module startup. This does the following
+ *
+ * Sets and persists the startupType on flows that have not had this saved previously
+ * Sets up wiretap triggers that have not been saved previously
+ */
+public class StartupModuleProcessor {
+
+    private final StartupModuleConfiguration configuration;
+
+    private final StartupControlDao startupControlDao;
+
+    private final JobAwareFlowEventListener jobAwareFlowEventListener;
+
+    public final static String WIRETAP_JOB = "wiretapJob";
+
+
+    private static final Logger logger = LoggerFactory.getLogger(StartupModuleProcessor.class);
+
+    public StartupModuleProcessor(StartupModuleConfiguration configuration, StartupControlDao startupControlDao,
+    JobAwareFlowEventListener jobAwareFlowEventListener){
+        this.configuration = configuration;
+        this.startupControlDao = startupControlDao;
+        this.jobAwareFlowEventListener = jobAwareFlowEventListener;
+    }
+
+    public void run(Module<Flow> module){
+        if (configuration.areStartupTypePropertiesPopulated()){
+            saveFlowsStartupTypeValueIfNotPreviouslySaved(module, configuration);
+        }
+        if (Boolean.TRUE.equals(configuration.getDeleteAllPreviouslySavedWiretaps())){
+            deleteAllPreviouslySavedWiretaps(module);
+        }
+        if (configuration.getWiretaps() != null){
+            savePreviouslyUnsavedWiretaps(module, configuration);
+        }
+    }
+
+    private void deleteAllPreviouslySavedWiretaps(Module<Flow> module) {
+        logger.info("Deleting all previously saved wiretaps");
+        List<Trigger> existingTriggers = jobAwareFlowEventListener.getTriggers();
+        if (!CollectionUtils.isEmpty(existingTriggers)) {
+            List<Trigger> wiretapTriggers = existingTriggers.stream().filter(t -> t.getModuleName().equals(module.getName())
+                && t.getJobName().equals(WIRETAP_JOB)).collect(Collectors.toList());
+            for (Trigger trigger : wiretapTriggers) {
+                logger.info("About to delete wiretap trigger [{}]", trigger);
+                jobAwareFlowEventListener.deleteDynamicTrigger(trigger.getId());
+            }
+        }
+    }
+
+    private void savePreviouslyUnsavedWiretaps(Module<Flow> module, StartupModuleConfiguration configuration) {
+        logger.info("Saving these wiretap triggers if havent been saved already [{}]", configuration.getWiretaps());
+        List<TriggerImpl> triggerList = configuration.getWiretaps().entrySet().stream().map(entry -> {
+            WiretapTriggerConfiguration wc = entry.getValue();
+            Map<String, String> params = new HashMap();
+            Integer timeToLive = Integer.parseInt(wc.getTimeToLiveInSeconds()) * 1000;
```

Comment:
Not sure of this conversion. I think TimeToLive on Wiretap is expressed in minutes not seconds. Please verify and confirm